### PR TITLE
feat(entire-app): This change allows the users to select the headers that they do not want Keploy to record

### PR DESCRIPTION
**Pull request by PranshuSrivastava**
Signed-off-by: Pranshu Srivastava <iampranshu24@gmail.com>

## Related Issue
  - Info about Issue or bug

Closes: #[issue number that will be closed through this PR]

#### Describe the changes you've made
A clear and concise description of what you have done to successfully close your assigned issue. Any new files? or anything you feel to let us know!

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |

### CODE REVIEW

**Review by slayerjain — CHANGES_REQUESTED** [review]
Please remove `protoc-3.15.8-linux-x86_64.zip` file from this.

**Comment by slayerjain** [comment]
@PranshuSrivastava The `grpc` package should be very lightweight. We shouldn't be putting this filtering logic there. The `grpc` package's purpose is to just provide the transport bindings for exposing the services inside the `pkg` as gRPC service.

File: go.mod [hunk 1]
```diff
@@ -2,7 +2,7 @@ module go.keploy.io/server
 
 go 1.17
 
-// replace github.com/keploy/go-sdk => github.com/iamskp99/go-sdk v0.0.6-alpha
+replace github.com/keploy/go-sdk => ../go-sdk/
```
Comment (re-Tick) [inline go.mod:5]: Please remove or comment the replace statement for local dependency

**Review by slayerjain — APPROVED** [review]
LGTM!

**Pull request merged**